GabijaBern/assignment1/Anagrams.java

```diff
@@ -0,0 +1,85 @@
+//Q1: Given two strings, determine if one is an anagram of the other.
+//Parts: 1)stringSensitive()
+//		 2)stringInsensitive()
+//		 3)sentences()
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Anagrams {
+
+	
+	//-----------------first part of the question------------------------------
+	public static boolean stringSensitive(String firstWord, String secondWord)
+	{
+		//we could first check if those words are the same length
+		if(firstWord.length()!=secondWord.length()) 
+			return false;
+		
+		Map<Character, Integer> firstMap = createCharMap(firstWord);
+		Map<Character, Integer> secondMap = createCharMap(secondWord);
+			
+		return(firstMap.equals(secondMap));
+	}
+	
+	//-------------------second part of the question-----------------------
+	public static boolean stringInsensitive(String firstWord, String secondWord)
+	{
+		//we don't want to distinguish lower and upper case letter, so we
+		//just turn all of them to lower
+		firstWord.toLowerCase();
+		secondWord.toLowerCase();
+		
+		return stringSensitive(firstWord, secondWord);
+	}
+	
+	//--------------third part of the question----------------------
+	public static boolean sentences(String firstSentence, String secondSentence)
+	{
+		String[] firstSentenceWords = firstSentence.split("\\W+");
+		String[] secondSentenceWords = secondSentence.split("\\W+");
+		
+		List<Map<Character, Integer>> firstListOfMaps = new ArrayList<Map<Character, Integer>>();
+		List<Map<Character, Integer>> secondListOfMaps = new ArrayList<Map<Character, Integer>>();
+		
+		if(firstSentenceWords.length==firstSentenceWords.length)

```

Comment:
This is another example of the case we discussed above: instead of nesting within the `if`-statement, you can negate the condition and return early.